pkgs/development/python-modules/netcdf4/default.nix

```diff
@@ -1,50 +1,72 @@
-{ stdenv, lib, buildPythonPackage, fetchPypi, isPyPy, pytest
+{ stdenv, lib, buildPythonPackage, fetchPypi, isPyPy
 , numpy, zlib, netcdf, hdf5, curl, libjpeg, cython, cftime
-}:
-buildPythonPackage rec {
-  pname = "netCDF4";
-  version = "1.5.3";
+,mpi4py ? null, openssh ? null}:
+assert (mpi4py != null) -> 
+  openssh != null
+  && hdf5.mpi == mpi4py.mpi 
+  && netcdf.mpi == mpi4py.mpi;
 
-  disabled = isPyPy;
+let 
+  mpiSupport = (mpi4py != null);
+  mpi= if mpiSupport then mpi4py.mpi else null;
+in
+  buildPythonPackage rec {
+    pname = "netCDF4";
+    version = "1.5.3";
+  
+    disabled = isPyPy;
+    
+    src = fetchPypi {
+      inherit pname version;
+      sha256 = "2a3ca855848f4bbf07fac366da77a681fcead18c0a8813d91d46302f562dc3be";
+    };
+  
+    checkInputs = [ netcdf numpy ] ++(if mpiSupport then [ mpi4py openssh mpi] else []);
```

Comment:
```suggestion
    checkInputs = [ netcdf numpy ] 
      ++ lib.optionals mpiSupport [ mpi4py openssh mpi ];
```

Reply:
I doubt that `openssh` and `mpi` need to be in the `checkInputs` in the first place.

Reply:
If I don't add  openssh I get the following error while running the tests:
-------------------------------------------------------------------------
The value of the MCA parameter "plm_rsh_agent" was set to a path
that could not be found:

  plm_rsh_agent: ssh : rsh

Please either unset the parameter, or check that the path is correct
--------------------------------------------------------------------------
[localhost:00218] [[INVALID],INVALID] FORCE-TERMINATE AT Not found:-13 - error plm_rsh_component.c(335)
[localhost:00217] [[INVALID],INVALID] ORTE_ERROR_LOG: Unable to start a daemon on the local node in file ess_singleton_module.c at line 716
[localhost:00217] [[INVALID],INVALID] ORTE_ERROR_LOG: Unable to start a daemon on the local node in file ess_singleton_module.c at line 172
--------------------------------------------------------------------------
It looks like orte_init failed for some reason; your parallel process is
likely to abort.  There are many reasons that a parallel process can
fail during orte_init; some of which are due to configuration or
environment problems.  This failure appears to be an internal failure;
here's some additional information (which may only be relevant to an
Open MPI developer):

  orte_ess_init failed
  --> Returned value Unable to start a daemon on the local node (-127) instead of ORTE_SUCCESS
--------------------------------------------------------------------------
--------------------------------------------------------------------------
It looks like MPI_INIT failed for some reason; your parallel process is
likely to abort.  There are many reasons that a parallel process can
fail during MPI_INIT; some of which are due to configuration or environment
problems.  This failure appears to be an internal failure; here's some
additional information (which may only be relevant to an Open MPI
developer):

  ompi_mpi_init: ompi_rte_init failed
  --> Returned "Unable to start a daemon on the local node" (-127) instead of "Success" (0)
--------------------------------------------------------------------------
*** An error occurred in MPI_Init_thread
*** on a NULL communicator
*** MPI_ERRORS_ARE_FATAL (processes in this communicator will now abort,
***    and potentially your MPI job)

but I removed mpi which was indeed a leftover